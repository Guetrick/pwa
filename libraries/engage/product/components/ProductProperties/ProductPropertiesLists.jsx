import React from 'react';
import PropTypes from 'prop-types';
import { getSubgroupsFromProperties } from './helpers/getSubgroupsFromProperties';
import { getPropertiesWithoutSubgroup } from './helpers/getPropertiesWithoutSubgroup';
import { getPropertiesBySubgroup } from './helpers/getPropertiesBySubgroup';
import ProductPropertiesGroup from './ProductPropertiesGroup';
import ProductPropertiesRows from './ProductPropertiesRows';

/**
 * Renders lists of properties that are eventually grouped.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const ProductPropertiesLists = ({ properties }) => (
  <React.Fragment>
    <ProductPropertiesRows properties={getPropertiesWithoutSubgroup(properties)} />
    {getSubgroupsFromProperties(properties).map(group => (
      <React.Fragment key={group}>
        <ProductPropertiesGroup group={group} />
        <ProductPropertiesRows properties={getPropertiesBySubgroup(properties, group)} />
      </React.Fragment>
    ))}
  </React.Fragment>
);

ProductPropertiesLists.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default ProductPropertiesLists;
