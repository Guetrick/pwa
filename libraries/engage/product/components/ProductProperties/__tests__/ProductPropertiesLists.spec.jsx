import React from 'react';
import { shallow } from 'enzyme';
import ProductPropertiesLists from '../ProductPropertiesLists';

const propertiesNoSubs = [
  {
    name: 'test1',
    label: 'Test 1',
  },
  {
    name: 'test2',
    label: 'Test 2',
  },
];

const propertiesSubs = [
  {
    name: 'test1',
    label: 'Test 1',
    subDisplayGroup: 'Test 1',
  },
  {
    name: 'test2',
    label: 'Test 2',
    subDisplayGroup: 'Test 2',
  },
];

describe('<ProductPropertiesLists />', () => {
  it('should only render simple rows', () => {
    const wrapper = shallow(<ProductPropertiesLists key="1" properties={propertiesNoSubs} />);
    expect(wrapper.find('ProductPropertiesRows').length).toEqual(1);
    expect(wrapper.find('ProductPropertiesGroup').length).toEqual(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render groups only', () => {
    const wrapper = shallow(<ProductPropertiesLists key="2" properties={propertiesSubs} />);
    expect(wrapper.find('ProductPropertiesRows').length).toEqual(3);
    expect(wrapper.find('ProductPropertiesRows').at(0).prop('properties')).toEqual([]);
    expect(wrapper.find('ProductPropertiesRows').at(1).prop('properties')).toEqual([
      {
        name: 'test1',
        label: 'Test 1',
        subDisplayGroup: 'Test 1',
      },
    ]);
    expect(wrapper.find('ProductPropertiesRows').at(2).prop('properties')).toEqual([
      {
        name: 'test2',
        label: 'Test 2',
        subDisplayGroup: 'Test 2',
      },
    ]);
    expect(wrapper.find('ProductPropertiesGroup').length).toEqual(2);
    expect(wrapper.find('ProductPropertiesGroup').at(0).prop('group')).toEqual('Test 1');
    expect(wrapper.find('ProductPropertiesGroup').at(1).prop('group')).toEqual('Test 2');
  });
});
