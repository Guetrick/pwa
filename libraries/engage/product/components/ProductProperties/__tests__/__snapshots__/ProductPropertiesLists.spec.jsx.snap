// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesLists /> should only render simple rows 1`] = `
<Fragment>
  <ProductPropertiesRows
    properties={
      Array [
        Object {
          "label": "Test 1",
          "name": "test1",
        },
        Object {
          "label": "Test 2",
          "name": "test2",
        },
      ]
    }
  />
</Fragment>
`;
